### ISSUE TITLE
bool() of and HTML field will never return False

### ISSUE THREAD

**Issue by LoisRForgeFlow** [odoo/odoo#69560]
Impacted versions: at least 13 and 14

Steps to reproduce:
Imagine you have a report that has this code:
```
<div t-if="doc.conditions">
    <strong><p>Term and conditions</p></strong>
    <span t-field="doc.conditions" />
</div>
```

and in that doc you have the html editor completely empty:
![image](https://user-images.githubusercontent.com/23449160/115422576-546a9100-a1fd-11eb-844d-256565042605.png)

Current behavior:
The report includes the div:
![image](https://user-images.githubusercontent.com/23449160/115423527-318cac80-a1fe-11eb-8237-b1de4e14f5a4.png)

Expected behavior:
div is not printed

More details:

What it is really stored in the database is `<p><br></p>` instead of a empty string.

Looking at the the frontend code I can see that this is something expected, because it is handled with the following function: https://github.com/odoo/odoo/blob/14.0/addons/web_editor/static/src/js/backend/field_html.js#L120.

```
isSet: function () {
    var value = this.value && this.value.split('&nbsp;').join('').replace(/\s/g, ''); // Removing spaces & html spaces
    return value && value !== "<p></p>" && value !== "<p><br></p>" && value.match(/\S/);
},
```

Wouldn't we want a similar function in the backed?

**Comment by pedrobaeza** [comment]
Your report should handled that situation. Although desirable, there's no such function for now AFAIK. Anyway, this can't be considered an Odoo bug. It will be if there's a core report that does that, but the fix will be to check for the other condition.

**Comment by LoisRForgeFlow** [comment]
I'm not saying it is a bug and I know I can handle it in my report. I'm just raising a question, maybe it is something that could be a good improvement for the framework, consider it a wishlist if you prefer.

**Comment by pedrobaeza** [comment]
Yes, I agree, but my experience with wishlists is that they are considered having a PR in the middle, not just a wishlist issue.

**Comment by LoisRForgeFlow** [comment]
That is right, I might do something, but I wanted to see some opinions before, maybe it is not as useful as I think.

**Comment by intero-chz** [comment]
It would be very useful.

Not even thinking about "new" empty HTML fields which will be filled with that crap on writes/updates on other fields...

**Comment by quentinDupont** [comment]
Hey ! @LoisRForgeFlow how do you handle it in your qweb report ? with a t-if != "<p><br></p>" ??
